### DISCUSSION TITLE
Printf's not working and Path to add new library files

### DISCUSSION THREAD

**Discussion by iamraoakash** [Lora-net/LoRaMac-node#1196] [Q&A]
Respected all,

I have a STM32 Nucleo L073 MCU and I want to use a I2C pressure sensor.
I have LoRaMAC 4.5.2 and according to the path it is build correctly as there aren't any errors as can be seen in the screenshot below. 
![image](https://user-images.githubusercontent.com/92906445/139570870-0240f110-7f01-4c88-9ac1-d99e51c12335.png)

Now as per the L073 board.c file the UART is initialized with 921600 baud rate, 8 Data bits, 1 Stop bit  and No Parity. 
![image](https://user-images.githubusercontent.com/92906445/139570938-162050a0-0a62-401a-8ba0-a7f3ab306a74.png)

and I am using Tera Term with the same configuration, it can be verified in the screenshot below:
![image](https://user-images.githubusercontent.com/92906445/139570993-d65b5453-694d-4512-8625-75c4cacac574.png)

Now as can be seen, when I upload the .bin file as it is built I see some things being printed on the terminal but the printf statements don't work as intended (as I have written it in the main.c file)
![image](https://user-images.githubusercontent.com/92906445/139571134-638bb41f-d73a-4d38-80f1-eb1dcc7baa62.png)

Please help me in figuring out as to where the mistake has been made from my end. 
Also I would like to know that if I want to add custom header and .c files at what location should I add the files?

**Comment by mluis1** [comment]
Please find at below link a tutorial on how to add an I2C BME280 Sensor support to this project.
https://lora-developers.semtech.com/learn/hands-on-labs/loramac-node-lab/loramac-node-lab-overview/
